src/lib.rs

```diff
@@ -97,3 +99,9 @@ pub trait GenFn4<T>: Fn(&PermutationTable, &Point4<T>) -> T {}
 impl<T, F> GenFn2<T> for F where F: Fn(&PermutationTable, &Point2<T>) -> T, {}
 impl<T, F> GenFn3<T> for F where F: Fn(&PermutationTable, &Point3<T>) -> T, {}
 impl<T, F> GenFn4<T> for F where F: Fn(&PermutationTable, &Point4<T>) -> T, {}
+
+pub trait Module<T> {
+    type Output;
+
+    fn get(&self, point: T) -> Self::Output;
+}
```

Comment:
Added. This let me remove the lifetime specifiers from Select, and remove some of the refs from the select example. The Perlin module still had to be specified by ref since it doesn't have a copy implementation.